include/zephyr/usb/usbd.h

```diff
@@ -195,68 +213,89 @@ struct usbd_cctx_vendor_req {
-	void (*feature_halt)(struct usbd_class_node *const node,
+	void (*feature_halt)(struct usbd_class_data *const c_data,
 			     uint8_t ep, bool halted);
 
 	/** Configuration update handler */
-	void (*update)(struct usbd_class_node *const node,
+	void (*update)(struct usbd_class_data *const c_data,
 		       uint8_t iface, uint8_t alternate);
 
 	/** USB control request handler to device */
-	int (*control_to_dev)(struct usbd_class_node *const node,
+	int (*control_to_dev)(struct usbd_class_data *const c_data,
 			      const struct usb_setup_packet *const setup,
 			      const struct net_buf *const buf);
 
 	/** USB control request handler to host */
-	int (*control_to_host)(struct usbd_class_node *const node,
+	int (*control_to_host)(struct usbd_class_data *const c_data,
 			       const struct usb_setup_packet *const setup,
 			       struct net_buf *const buf);
 
 	/** Endpoint request completion event handler */
-	int (*request)(struct usbd_class_node *const node,
+	int (*request)(struct usbd_class_data *const c_data,
 		       struct net_buf *buf, int err);
 
 	/** USB power management handler suspended */
-	void (*suspended)(struct usbd_class_node *const node);
+	void (*suspended)(struct usbd_class_data *const c_data);
 
 	/** USB power management handler resumed */
-	void (*resumed)(struct usbd_class_node *const node);
+	void (*resumed)(struct usbd_class_data *const c_data);
 
 	/** Start of Frame */
-	void (*sof)(struct usbd_class_node *const node);
+	void (*sof)(struct usbd_class_data *const c_data);
 
 	/** Class associated configuration is selected */
-	void (*enable)(struct usbd_class_node *const node);
+	void (*enable)(struct usbd_class_data *const c_data);
 
 	/** Class associated configuration is disabled */
-	void (*disable)(struct usbd_class_node *const node);
+	void (*disable)(struct usbd_class_data *const c_data);
 
 	/** Initialization of the class implementation */
-	int (*init)(struct usbd_class_node *const node);
+	int (*init)(struct usbd_class_data *const c_data);
 
 	/** Shutdown of the class implementation */
-	void (*shutdown)(struct usbd_class_node *const node);
+	void (*shutdown)(struct usbd_class_data *const c_data);
+
+	/** Get function descriptor based on speed parameter */
+	void *(*get_desc)(struct usbd_class_data *const c_data,
+			  const enum usbd_speed speed);
 };
 
 /**
  * @brief USB device support class data
  */
 struct usbd_class_data {
+	/** Name of the USB device class instance */
+	const char *name;
 	/** Pointer to USB device stack context structure */
 	struct usbd_contex *uds_ctx;
-	/** Pointer to a class implementation descriptor that should end with
-	 *  a nil descriptor (bLength = 0 and bDescriptorType = 0).
-	 */
-	void *desc;
+	/** Pointer to device support class API */
+	const struct usbd_class_api *api;
 	/** Supported vendor request table, can be NULL */
 	const struct usbd_cctx_vendor_req *v_reqs;
+	/** Pointer to private data */
+	void *priv;
+};
+
+/**
+ * @cond INTERNAL_HIDDEN
```

Comment:
It turns out that `/** @cond INTERNAL_HIDDEN */` must have a matching `/** @endcond */` as otherwise the build fails. Here the endcond should be right after `struct usbd_class_node`.